Enable SEPA IBAN field for UPE checkout

cesarcosta99 opened the pull request:
# Changes proposed in this Pull Request:

Fixes #1751 

This PR adds a logic to mount the SEPA IBAN field in the new checkout experience for both checkout and adding a payment method flows.

# Testing instructions
1. Ensure you have the UPE feature flag set to `yes` in the database and SEPA enabled in the store.
2. Visit http://localhost:8082/ and log in.
3. Go to `My account` > `Payment methods` > `Add payment method`.
4. You should see an IBAN field when adding a SEPA payment method.
5. Go to `Shop` and add an item to the cart.
6. Go to Checkout and select `SEPA Direct Debit`.
7. You should see an IBAN field to type in.


-------------------
- [x] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [ ] Did you make changes, or create a **new .js file**? If **Gruntfile.js** exists in the repo, make sure to run `grunt`.

cesarcosta99 committed:
Add SEPA element options to be sent in server data

cesarcosta99 committed:
Add iban mount to UPE checkout

EvanDotPro commented on client/classic/upe/index.js:
```diff
@@ -352,6 +362,13 @@ jQuery( function ( $ ) {
+			! $( '#stripe-iban-element' ).children().length
+		) {
+			iban.mount( '#stripe-iban-element' );
+		}
```
This is a bit pedantic, especially in this case, but since this exact conditional is used >1 time, I'd maybe consider moving it into a function and calling that for the sake of future maintainability. That way if the condition ever needs to be adjusted in the future for example, there's only a single instance to change— less opportunity for human error / accidental omissions.

To be clear, this is not critical and not a blocker for approval.

cesarcosta99 committed:
Refactor IBAN mount check condition

Add a new doesIbanNeedToBeMounted function to keep only a single instance and make it better for future maintainability.
